src/pudl/transform/ferc1.py

```diff
@@ -519,28 +520,25 @@ class Ferc1AbstractTableTransformer(AbstractTableTransformer):
     the transformed data.
     """
 
-    xbrl_metadata_json: list[dict] = []
-    """An array of JSON objects extracted from the FERC 1 XBRL taxonomy."""
-
-    xbrl_metadata_normalized: pd.DataFrame = pd.DataFrame()
-    """A semi-normalized dataframe containing table-specific XBRL metadata."""
+    xbrl_metadata: pd.DataFrame
+    """Dataframe combining XBRL metadata for both instant and duration table columns."""
```

Comment:
Dataframe **of** XBRL metadata for both instant and duration table columns?

Reply:
why does this need to be here as well as defined in `__init__`?

Reply:
I think it's generally better to define what attributes a class has up front than to randomly stash stuff in a class attribute you make up as you go along.  This is a statement about the kinds of things one should expect to find inside of an object of type `Ferc1AbstractTableTransformer` (which allows it to be documented as part of the class) and also allows tab-completion based on the declared type.

I should set a default value of an empty dataframe too.

Reply:
okay!!